pkg/controller/multi_writer_at.go

```diff
@@ -45,18 +47,26 @@ func (m *MultiWriterAt) WriteAt(p []byte, off int64) (n int, err error) {
 			if err != nil {
```

Comment:
```go
deffer wg.Done()
n, err := w.WriteAt(p, off)
errors[index] = err
written[index] = n
```

Now you don't need to keep the state variables and this pattern also doesn't require any syncronisation.